lib/uefi/boot_service_provider.cpp

```diff
@@ -0,0 +1,380 @@
+
+#include "boot_service_provider.h"
+#include "boot_service.h"
+#include "kernel/vm.h"
+#include "lib/dlmalloc.h"
+#include "types.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+EfiStatus unload(EfiHandle handle) { return SUCCESS; }
+
+bool guid_eq(const EfiGuid *a, const EfiGuid *b) {
+  return memcmp(a, b, sizeof(*a)) == 0;
+}
+
+bool guid_eq(const EfiGuid *a, const EfiGuid &b) {
+  return memcmp(a, &b, sizeof(*a)) == 0;
+}
+
+vmm_aspace_t *get_address_space() {
```

Comment:
suggestion: a more descriptive name such as 'set boot aspace' would be clearer what it's actually doing, since it's modifying the existing aspace.

Reply:
Done